src/main/java/de/zalando/aruha/nakadi/validation/EventBodyMustRespectSchema.java

```diff
@@ -169,7 +165,22 @@ public JSONSchemaValidator(final JSONObject effectiveSchema) {
 
             return Optional.empty();
         } catch (final ValidationException e) {
-            return Optional.of(new ValidationError(e.getMessage()));
+            StringBuilder builder = new StringBuilder();
+            collectErrorMessages(e, builder);
+
+            return Optional.of(new ValidationError(builder.toString()));
+        }
+    }
+
+    private void collectErrorMessages(ValidationException e, StringBuilder builder) {
+        builder.append(e.getMessage());
+
+        if (!e.getCausingExceptions().isEmpty()) {
```

Comment:
Good catch :+1: done in https://github.com/zalando/nakadi/commit/f39d94918c1371a3f943688a29445aaf94cdcbe2